lib/types/src/circleci.ts

```diff
@@ -22,10 +23,42 @@ export type Workflow = {
 }
 
 export interface CircleConfig {
-  version: number
-  orbs: { [orb: string]: string }
-  workflows?: {
-    version: number
-    [workflow: string]: Workflow | number
+  version: 2.1
```

Comment:
**suggestion:** I think this type is only used in the `circleci-config.ts` file, where [`PartialDeep` is applied to it ](https://github.com/Financial-Times/dotcom-tool-kit/pull/359/files#diff-f157d7392cc3db08e0c5b213317a296767e475221a5be30ae3bcc8b94fa7613fR25) which makes every field optional. Could you also add a comment in this file to note that actually every field is optional (and how/where that's done) so that it's not confusing to someone reviewing the types in isolation

Reply:
Seeing as it's aliased to `CircleCIState` it is used in quite a few places in `circleci-config.ts` where `PartialDeep` is not applied (`PartialDeep` is only applied when using `CircleCIStatePartial` instead). So the fields are not optional unless you're using the `CircleCIStatePartial` type instead of `CircleCIState` or `CircleConfig`

Reply:
Ahh I seeeeee